homework_oop/src/homework_oop/Computer.java

```diff
@@ -0,0 +1,121 @@
+package homework_oop;
+
+import java.util.Arrays;
+
+/**
+ * Represents a computer, described with name and MAC addres
+ * MAC addres is defined with nested class MacAddress
+ * @author ajla
+ *
+ */
+public abstract class Computer {
+	// Proper MAC address will have exactly 12 characters, without separators
+	private static final int MAC_ADDRESS_LENGTH = 12;
+	private String name;
+	private MacAddress macAddress;
+	
+	/**
+	 * Constructor, sets Computer parameters, 
+	 * @param name
+	 * @param macAdress
+	 */
+	public Computer(String name, char[] macAddress) {
+		super();
+		this.name = name;
+		this.macAddress = new MacAddress(macAddress);
+	}
+
+	/**
+	 * @return the name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * @param name the name to set
+	 */
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	/**
+	 * @return the macAdress
+	 */
+	public MacAddress getMacAdress() {
+		return macAddress;
+	}
+
+	/**
+	 * @param macAdress the macAdress to set
+	 */
+	public void setMacAdress(MacAddress macAddress) {
+			this.macAddress = macAddress;		
+	}
+	
+	/**
+	 * Nested Class MacAddress
+	 * represents MAC address of the computer
+	 * It is given as array of characters
+	 * @author ajla
+	 *
+	 */
+	public class MacAddress {
+		private char[] address = new char[MAC_ADDRESS_LENGTH];
```

Comment:
this should not be allocated by default since constructor sets the value with the incoming argument.